run/env.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package run
+
+import (
+	"os"
+	"sort"
+
+	"github.com/mineiros-io/terramate/errors"
+	"github.com/mineiros-io/terramate/hcl"
+	"github.com/mineiros-io/terramate/stack"
+	"github.com/rs/zerolog/log"
+	"github.com/zclconf/go-cty/cty"
+)
+
+const (
+	// ErrParsingCfg indicates that an error happened while parsing configuration.
+	ErrParsingCfg errors.Kind = "parsing run env cfg"
+
+	// ErrLoadingGlobals indicates that an error happened while loading globals.
+	ErrLoadingGlobals errors.Kind = "loading globals to evaluate run env cfg"
+
+	// ErrEval indicates that an error happened while evaluating one of the
+	// terramate.config.run.env attributes.
+	ErrEval errors.Kind = "evaluating terramate.config.run.env attribute"
+
+	// ErrInvalidEnvVarType indicates the env var attribute
+	// has an invalid type.
+	ErrInvalidEnvVarType errors.Kind = "invalid env var type"
+)
+
+// EnvVars represents a set of environment variables to be used
+// when running commands. Each string follows the same format used
+// on os.Environ and can be used to set env on exec.Cmd.
+type EnvVars []string
+
+// Env will load environment variables to be exported when running any command
+// inside the given stack. The order of the env vars is guaranteed to be the same
+// and is ordered lexicographically.
```

Comment:
This is not a hard requirement for setting the env on the command, but it is useful for testing + Im planning on have an experimental command to show the environment vars for each stack, that will make the output nice/deterministic there too.